drivers/axi_core/axi_dac_core/axi_dac_core.c

```diff
@@ -877,6 +871,45 @@ int32_t axi_dac_init(struct axi_dac **dac_core,
 	return FAILURE;
 }
 
+int32_t axi_dac_data_setup(struct axi_dac *dac)
+{
+	struct axi_dac_channel *chan;
+	uint32_t i;
+
+	for (i = 0; i < dac->num_channels; i++) {
+		chan = &dac->channels[i];
+		if (chan) {
```

Comment:
Possible undefined behavior. Can be solved with double pointer or adding an initialized channel mask.

Reply:
You are right. Will check for best solution.